src/Requests/Request.php

```diff
@@ -79,21 +59,41 @@ abstract public function get();
      *
      * @return array
      *
-     * @throws PageNotFoundException
-     * @throws RequestException
      * @throws ClientExceptionInterface
+     * @throws ApiErrorException
      */
     public function send(): array
     {
-        $request = $this->message->createRequest('GET', $this->url . $this->getEndpoint());
-        $result = null;
+        $request = new GuzzleRequest('GET', $this->getEndpoint());
+        $requestResponse = $this->client->sendRequest($request);
+
+        $response = json_decode($requestResponse->getBody(), true, 512, JSON_BIGINT_AS_STRING);
+
+        if ($this->hasError($response)) {
+            $message = $response['descriptor'] ?? $response['response'];
+
+            throw new ApiErrorException($message, $requestResponse->getStatusCode());
+        }
+
+        return $response;
+    }
+
+    /**
+     * Check if the response contains an error.
+     *
+     * @param $response
+     * @return bool
+     */
+    protected function hasError($response): bool
+    {
+        if (! isset($response['error'])) {
```

Comment:
With the current styling, we do not put a space after an exclamation mark (e.g., the constructor of the Ban model).